src/main/java/by/onliner/listener/TestListener.java

```diff
@@ -0,0 +1,185 @@
+        logger.info("Test [" + result.getName() + "] FAILED");
+        try {
+            Recorder.stopRecording(result.getName());
+            appendLogToAllure();
+            takeScreenshot();
+            attachVideoToAllure(Recorder.savePath + result.getName());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /***
+     * When Test case get Skipped, this method is called.
+     */
+    @Override
+    public void onTestSkipped(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] SKIPPED");
+        try {
+            Recorder.stopRecording(result.getName());
+            appendLogToAllure();
+            takeScreenshot();
+            attachVideoToAllure(Recorder.savePath + result.getName());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void onTestFailedButWithinSuccessPercentage(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] FAILED but with some percentage of success");
+        try {
+            Recorder.stopRecording(result.getName());
+            appendLogToAllure();
+            takeScreenshot();
+            attachVideoToAllure(Recorder.savePath + result.getName());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void onTestFailedWithTimeout(ITestResult result) {
+        logger.info("Test [" + result.getName() + "] FAILED due to the timeout");
+        try {
+            Recorder.stopRecording(result.getName());
+            appendLogToAllure();
+            takeScreenshot();
+            attachVideoToAllure(Recorder.savePath + result.getName());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /***
+     * Attachment logging for report.
+     */
+    @Attachment(value = "Test logs", type = "text/plain")
+    private byte[] appendLogToAllure() {
+        try {
+            return FileUtils.readFileToByteArray(new File("target/logs/appTest.log"));
+        } catch (IOException ignored) {
+            return null;
+        }
+    }
+
+    /***
+     * Attachment screenshot for report.
+     */
+    @Attachment(value = "Screenshot", type = "image/png")
+    private byte[] takeScreenshot() {
+        return ((TakesScreenshot) DriverManager.getDriver()).getScreenshotAs(OutputType.BYTES);
+    }
+
+    /***
+     * Attachment video of test for report.
+     */
+    @Attachment(value = "VideoFile", type = "video/mp4")
+    public byte[] attachVideoToAllure(String videoFileLocation) {
+        try {
+            return Files.toByteArray(new File(videoFileLocation + ".mp4"));
+        } catch (IOException e) {
+            logger.debug("attachVideoToAllure(): FAILED\n" + e.getMessage());
+            return new byte[0];
+        }
+    }
+
+    /***
+     * Clear log file after attaching log file to report.
+     */
+    private void clearFile() throws IOException {
```

Comment:
is clearFile part of TestListener functionality? Can you think of situations where you might need to clear any other file?

Reply:
When this funktionality will need in other methods, i will create Clear file Util

Reply:
it's better to do it now, you never know when you will need it. 
According to your logic, you should store everything what used once in one class but it breaks logic doesn't it?